torch/csrc/jit/codegen/cuda/kernel_ir.h

```diff
@@ -365,7 +372,7 @@ class TORCH_CUDA_CU_API Expr : public Node {
   // TODO(kir): revisit scope/nesting data structures
   Scope* scope_ = nullptr;
 
-  Bool* predicate_ = nullptr;
+  Predicate* predicate_ = nullptr;
```

Comment:
Can `Predicate` also include the `thread_predicate_` value of `GridReduction` and `GridWelford` so that we don't need to maintain them separately?

Reply:
Do you know why `thread_predicate_` is separate for `GridReduction` and `GridWelford`?

Reply:
IIRC, that's because it's used to generate boolean template parameters, e.g., `X/Y/Z_THREAD` of `gridReduce`.